lib/fs-writable.js

```diff
@@ -0,0 +1,162 @@
+// Copyright Node.js contributors. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+'use strict';
+
+const fs = require('fs');
+
+const Writable = require('./writable');
+const {
+  ERR_INVALID_ARG_TYPE,
+  ERR_OUT_OF_RANGE
+} = require('./errors');
+
+const openSymbol = Symbol('open');
+const errorSymbol = Symbol('error');
+const closeStreamSymbol = Symbol('closeStream');
+
+class WritableFileStream extends Writable {
+  constructor(path, {
+    fd = null,
+    start,
+    flags = 'w',
+    bufferSize,
+    mode = 0o666,
+    encoding = null,
+    autoClose = true,
+  } = {}) {
+    super({ bufferSize });
+
+    this.fd = fd;
+    this.path = path;
+
+    this.closed = false;
+    this.bytesWritten = 0;
+
+    this.autoClose = autoClose;
+
+    if (typeof path !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('path', 'string', path);
+    }
+
+    if (start !== undefined) {
+      if (typeof start !== 'number') {
+        throw new ERR_INVALID_ARG_TYPE('start', 'number', start);
+      }
+      if (start < 0) {
+        throw new ERR_OUT_OF_RANGE('start', '>= 0', `{start: ${start}}`);
+      }
+
+      this.pos = start;
+    }
+
+    if (encoding) this.setDefaultEncoding(encoding);
+
+    if (this.fd === null) this[openSymbol](flags, mode);
+  }
+
+  [errorSymbol](error, cb) {
+    if (cb) cb(error);
+    if (this.autoClose) {
+      this.destroy(error);
+    } else {
+      this.emit('error', error);
+      this.emit('close');
+    }
+  }
+
+  [openSymbol](flags, mode) {
+    fs.open(this.path, flags, mode, (error, fd) => {
+      if (error) {
+        this[errorSymbol](error);
+        return;
+      }
+
+      this.fd = fd;
+      this.emit('open', fd);
+    });
+  }
+
+  close(cb) {
```

Comment:
There is no such method in either WritableStream (Node.js) or the `Writable` below, I don't think it should be here. Or perhaps it should be `end` (this will require a signature change though)?